src/analysis/normed_space/compact_operator.lean

```diff
@@ -0,0 +1,430 @@
+⟨-K, hK.neg, mem_of_superset hKf $ λ x (hx : f x ∈ K), set.neg_mem_neg.mpr hx⟩
+
+lemma is_compact_operator.sub [topological_add_group M₄] {f g : M₁ → M₄}
+  (hf : is_compact_operator f) (hg : is_compact_operator g) : is_compact_operator (f - g) :=
+by rw sub_eq_add_neg; exact hf.add hg.neg
+
+variables (σ₁₄ M₁ M₄)
+
+/-- The submodule of compact continuous linear maps. -/
+def compact_operator [module R₁ M₁] [module R₄ M₄] [has_continuous_const_smul R₄ M₄]
+  [topological_add_group M₄] :
+  submodule R₄ (M₁ →SL[σ₁₄] M₄) :=
+{ carrier := {f | is_compact_operator f},
+  add_mem' := λ f g hf hg, hf.add hg,
+  zero_mem' := is_compact_operator_zero,
+  smul_mem' := λ c f hf, hf.smul c }
+
+end operations
+
+section comp
+
+variables {R₁ R₂ R₃ : Type*} [semiring R₁] [semiring R₂] [semiring R₃] {σ₁₂ : R₁ →+* R₂}
+  {σ₂₃ : R₂ →+* R₃} {M₁ M₂ M₃ : Type*} [topological_space M₁] [topological_space M₂]
+  [topological_space M₃] [add_comm_monoid M₁] [module R₁ M₁]
+
+lemma is_compact_operator.comp_clm [add_comm_monoid M₂] [module R₂ M₂] {f : M₂ → M₃}
+  (hf : is_compact_operator f) (g : M₁ →SL[σ₁₂] M₂) :
+  is_compact_operator (f ∘ g) :=
+begin
+  have := g.continuous.tendsto 0,
+  rw map_zero at this,
+  rcases hf with ⟨K, hK, hKf⟩,
+  exact ⟨K, hK, this hKf⟩
+end
+
+lemma is_compact_operator.continuous_comp {f : M₁ → M₂} (hf : is_compact_operator f) {g : M₂ → M₃}
+  (hg : continuous g) :
+  is_compact_operator (g ∘ f) :=
+begin
+  rcases hf with ⟨K, hK, hKf⟩,
+  refine ⟨g '' K, hK.image hg, mem_of_superset hKf _⟩,
+  nth_rewrite 1 preimage_comp,
+  exact preimage_mono (subset_preimage_image _ _)
+end
+
+lemma is_compact_operator.clm_comp [add_comm_monoid M₂] [module R₂ M₂] [add_comm_monoid M₃]
+  [module R₃ M₃] {f : M₁ → M₂} (hf : is_compact_operator f) (g : M₂ →SL[σ₂₃] M₃) :
+  is_compact_operator (g ∘ f) :=
+hf.continuous_comp g.continuous
+
+end comp
+
+section cod_restrict
+
+variables {R₁ R₂ : Type*} [semiring R₁] [semiring R₂] {σ₁₂ : R₁ →+* R₂}
+  {M₁ M₂ : Type*} [topological_space M₁] [topological_space M₂]
+  [add_comm_monoid M₁] [add_comm_monoid M₂] [module R₁ M₁] [module R₂ M₂]
+
+lemma is_compact_operator.cod_restrict {f : M₁ → M₂} (hf : is_compact_operator f)
+  {V : submodule R₂ M₂} (hV : ∀ x, f x ∈ V) (h_closed : is_closed (V : set M₂)):
+  is_compact_operator (set.cod_restrict f V hV) :=
+let ⟨K, hK, hKf⟩ := hf in
+⟨coe ⁻¹' K, (closed_embedding_subtype_coe h_closed).is_compact_preimage hK, hKf⟩
+
+end cod_restrict
+
+section restrict
+
+variables {R₁ R₂ R₃ : Type*} [semiring R₁] [semiring R₂] [semiring R₃] {σ₁₂ : R₁ →+* R₂}
+  {σ₂₃ : R₂ →+* R₃} {M₁ M₂ M₃ : Type*} [topological_space M₁] [uniform_space M₂]
+  [topological_space M₃] [add_comm_monoid M₁] [add_comm_monoid M₂] [add_comm_monoid M₃]
+  [module R₁ M₁] [module R₂ M₂] [module R₃ M₃]
+
+/-- If a compact operator preserves a closed submodule, its restriction to that submodule is
+compact.
+
+Note that, following mathlib's convention in linear algebra, `restrict` design the restriction of
```

Comment:
```suggestion
Note that, following mathlib's convention in linear algebra, `restrict` means the restriction of
```
I don't think "désigne" in the French sense works in English.

Reply:
Ah, indeed.  You can say "designates" if you want to keep the parallel.

Reply:
I'm surprised that I wrote that, it sounds weird even to me :sweat_smile: